spacescout_labstats/test/test_cte_techloan.py

```diff
@@ -0,0 +1,275 @@
+        # Mock a get call with different status codes returned
+        for status in status_codes:
+            mock = Mock()
+            mock.status_code = status
+            mock.json = Mock(return_value=test_json)
+
+            with patch.object(requests, 'get', return_value=mock):
+                techloan_data = cte_techloan.get_techloan_data()
+                if (status == 200):
+                    self.assertEqual(techloan_data, test_json)
+                else:
+                    self.assertIs(techloan_data, None)
+
+        # Mock a call with invalid techloan_data
+        test_json = 1
+        mock = Mock()
+        mock.status_code = 200
+        mock.json = Mock(return_value=test_json)
+
+        with patch.object(requests, 'get', return_value=mock):
+            techloan_data = cte_techloan.get_techloan_data()
+            self.assertIs(techloan_data, None)
+
+    def test_get_endpoint_data(self):
+        """
+        Tests that item availability is retrieved and updated in the spaces
+        appropriately
+        """
+        test_json = self.load_json_file('cte_techloan.json')
+        test_json_data = self.load_json_file('cte_techloan_type_data.json')
+        # Test getting endpoint data with valid spaces and space data succeeds
+        with patch.object(cte_techloan, 'get_techloan_data',
+                          return_value=test_json_data):
+            cte_techloan.get_endpoint_data(test_json)
+        # Test getting endpoint data with invalid data returns None
+        with patch.object(cte_techloan, 'get_techloan_data',
+                          return_value=None):
+            self.assertIs(cte_techloan.get_endpoint_data(test_json), None)
+
+    @override_settings(CTE_TECHLOAN_URL="xxx")
+    def test_get_endpoint_data_no_setting(self):
+        """
+        Tests that an exception is raised when getting endpoint
+        data without a CTE_TECHLOAN_URL in settings
+        """
+        del settings.CTE_TECHLOAN_URL
+        with self.assertRaises(Exception):
+            cte_techloan.get_endpoint_data(test_json)
+
+    def test_load_techloan_data_into_spaces(self):
+        """
+        Tests that techloan data is loaded into the corresponding
+        items within a space that match the type_id of the data
+        """
+        # Initial test with matching cte_type_id in data and spaces
+        test_spaces = self.load_json_file('cte_techloan.json')
+        test_data = self.load_json_file('cte_techloan_type_data.json')
+        cte_techloan.load_techloan_data_into_spaces(test_spaces, test_data)
+        self.assertEqual(
+            test_spaces[0]["items"][0]["extended_info"]["i_is_active"],
+            'true')
+
+        # Test with a non-matching cte_type_id in space item
+        test_spaces = self.load_json_file('cte_techloan.json')
+        test_spaces[0]["items"][0]["extended_info"]["cte_type_id"] = 22
+        cte_techloan.load_techloan_data_into_spaces(test_spaces, test_data)
+
```

Comment:
It would be good to assert that item did not get data inserted into it here. Maybe get the default data before running calling load_techloan_data_into_spaces and assert that it matches?